src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.ProjectFormat/TryConvertProjectConverterStep.cs

```diff
@@ -39,7 +39,9 @@ public class TryConvertProjectConverterStep : UpgradeStep
             _runner = runner ?? throw new ArgumentNullException(nameof(runner));
         }
 
-        protected override bool IsApplicableImpl(IUpgradeContext context) => context?.CurrentProject is not null;
+        protected override bool IsApplicableImpl(IUpgradeContext context) => context?.CurrentProject is not null
+            /* 03/02/2020 - try-convert version 0.7.212201 does not support the migration of Visual Basic WPF applications */
+            && !(context.CurrentProject.Language == Languages.VisualBasic && (context.CurrentProject.Components & ProjectComponents.WPF) == ProjectComponents.WPF);
```

Comment:
I'd rather not disable the TryConvertProjectConverter step for VB WPF projects because users will (reasonably) expect it to work and be confused about it not showing up in the list.

I think we should only return false from `IsApplicableImpl` if there's a good reason (not a missing feature or a bug) why the step doesn't apply.

In this case, I'd rather check for VB WPF during InitializeImplAsync and then (1) give a warning so the user understands that we wanted to run try-convert but couldn't because it's not supported yet and (2) return a 'skipped' result so that it's obvious that the try-convert step was not executed but isn't blocking forward progress.

Reply:
Also, you can check for components more easily with [HasFlags](https://docs.microsoft.com/en-us/dotnet/api/system.enum.hasflag?view=net-5.0).

Reply:
After our conversation, let's get rid of this and just do a readiness check.

Reply:
As discussed the TryConvertProjectConverter step will fail the readiness check. The readiness check will own responsibility for providing the reason of failure to the user.